src/main/java/view/HomePageUI/TopRightIconsPanel.java

```diff
@@ -50,24 +51,27 @@ public TopRightIconsPanel(JFrame parentFrame) {
                         );
                     }
                 });
+            }
+            if ("📩".equals(icon)) {
+                iconButton.addActionListener(e -> {
+                    String sectionFilter = resolveSectionFilter(parentFrame);
+                    if (sectionFilter != null) {
+                        ChatPage chatpage = new ChatPage(parentFrame, currentuser);
```

Comment:
the action listener follows a good coding style, it is clean and tidy.

Reply:
There are two constructors, one using parentFrame and another using both parentFrame and currentuser. If the first constructor is not being used, consider removing it to simplify the code.